### PR TITLE
Add a new option 'check-str-concat-over-line-jumps' to check 'implicit-str-concat-in-sequence'

### PR DESCRIPTION
**Pull request by Lucas-C**
Adding the option discussed in https://github.com/PyCQA/pylint/pull/1655

### CODE REVIEW

File: pylint/checkers/strings.py [hunk 1]
```diff
@@ -601,22 +615,26 @@ def process_tokens(self, tokens):
...
+                while j < len(tokens) and tokens[j].type in (tokenize.NEWLINE, tokenize.NL, tokenize.COMMENT):
+                    j += 1
+                next_token = tokens[j] if j < len(tokens) else None
                 if encoding != "ascii":
                     # We convert `tokenize` character count into a byte count,
                     # to match with astroid `.col_offset`
                     start = (start[0], len(line[: start[1]].encode(encoding)))
                 self.string_tokens[start] = (str_eval(token), next_token)
 
-    @check_messages(*(MSGS.keys()))
```
Comment (PCManticore) [inline pylint/checkers/strings.py]: Oh, thanks for catching that!

**Comment by PCManticore** [comment]
By the way @Lucas-C Can we check the CI passing here? it currently fails in the formatting step, which means that the test themselves don't actually get run. Also if you can rebase/merge master into your branch that would be helpful, as the master should be green right now.

**Comment by Lucas-C** [comment]
`black` executed locally & rebase done, this should be good to merge

Also, should I had some user documentation somewhere ?

**Comment by PCManticore** [comment]
@Lucas-C You can add an additional note in `docs/whatsnew/2.3.rst` which currently does not exist. Other than that, this should already appear in the documentation automatically under http://pylint.pycqa.org/en/latest/technical_reference/features.html

**Comment by Lucas-C** [comment]
Awesome than this is good to be merged for me !

**Comment by PCManticore** [comment]
Thanks a lot @Lucas-C !

**Pull request merged**